libcrux-intrinsics/src/avx2.rs

```diff
@@ -13,6 +13,7 @@ pub type Vec256Float = __m256;
 #[hax_lib::opaque]
 #[inline(always)]
 pub fn mm256_storeu_si256_u8(output: &mut [u8], vector: Vec256) {
+    #[cfg(not(hax))]
     debug_assert_eq!(output.len(), 32);
```

Comment:
Is this because hax does not yet handle debug asserts well? If so can we point to an issue here?

Reply:
It's actually because hax does handle debug asserts and translate them: we end up with a lot of garbadge code that is useless for verification.

Reply:
I added documentation to explain exactly why those are not wanted in hax